.history-03

```diff
@@ -0,0 +1,271 @@
+git checkout -b kubernetes-networks
+# Plan:
+# test web-app
+#  pod healthchecks
+#  deployments
+#  clusterip
+#  ipvs
+# ingress access
+#  metallb l2
+#  loadbalancer
+#  ingress-controller
+#  ingress rules
+
+# add healthchecks
+vim kubernetes-intro/web-pod.yaml
+...
+readinessProbe:
+  httpGet:
+    path: /index.html
+    port: 80
+...
+
+kubectl delete pods web
+kubectl apply -f kubernetes-intro/web-pod.yaml
+kubectl get pods # make sure it's in Running state
+kubectl describe pod/web # look for conditions and events
+
+# add liveness probe despite wrong port
+vim kubernetes-intro/web-pod.yaml
+...
+livenessProbe:
+  tcpSocket:
+    port: 8000
+
+kubectl delete pods web --force --grace-period=0
+kubectl apply -f kubernetes-intro/web-pod.yaml
+
+# questions:
+1. what's wrong with the following?
+```sh
+livenessProbe:
+  exec:
+command:
+      - 'sh'
+      - '-c'
+      - 'ps aux | grep my_web_server_process'
+```
+ - checking for main process health is dumb. Make it exit in case of errors.
```

Comment:
Yep, it is. Also, such probe requires `-o pipefail` to pass exit code from `grep` to upstream `sh` process